### PR TITLE
fixed nette 2.4 compatibility

### PR DESCRIPTION
**Pull request by matej21**
This change is [Reviewable](https://reviewable.io/reviews/kdyby/translation/117)

### CODE REVIEW

File: src/Kdyby/Translation/DI/TranslationExtension.php [hunk 1]
```diff
@@ -264,15 +264,14 @@ public function beforeCompile()
...
 		$registerToLatte = function (Nette\DI\ServiceDefinition $def) {
-			$def
-				->addSetup('?->onCompile[] = function($engine) { Kdyby\Translation\Latte\TranslateMacros::install($engine->getCompiler()); }', ['@self'])
-				->addSetup('addFilter', ['translate', [$this->prefix('@helpers'), 'translate']]);
+			$def->addSetup('?->onCompile[] = function($engine) { Kdyby\Translation\Latte\TranslateMacros::install($engine->getCompiler()); }', ['@self']);
 
 			if (method_exists('Latte\Engine', 'addProvider')) { // Nette 2.4
-				$def->addSetup('addProvider', ['translator', '@Kdyby\Translation\Translator']);
-
+				$def->addSetup('addProvider', ['translator', [$this->prefix('@helpers'), 'getTranslator']])
```
Comment (fprochazka) [inline src/Kdyby/Translation/DI/TranslationExtension.php]: Hmm, why not use the service directly? `$this->prefix('@default')` ?

Comment (matej21) [inline src/Kdyby/Translation/DI/TranslationExtension.php]: I'm [not sure](https://github.com/Kdyby/Translation/commit/7ae78c74e85b8a2966d807d30f5017f6f2a8aea7#diff-0614d9c5b2a16c577483bacde0b13568R160) :))

**Comment by JanTvrdik** [comment]
ping

**Comment by JanTvrdik** [comment]
double ping!

**Comment by Attanon** [comment]
ping

**Comment by Salamek** [comment]
...ping

**Comment by clovnrian** [comment]
ping

**Comment by fprochazka** [comment]
@matej21 thank you

**Pull request merged**